ports/stm32/rtc.c

```diff
@@ -577,7 +577,7 @@ mp_obj_t pyb_rtc_datetime(size_t n_args, const mp_obj_t *args) {
             mp_obj_new_int(2000 + date.Year),
             mp_obj_new_int(date.Month),
             mp_obj_new_int(date.Date),
-            mp_obj_new_int(date.WeekDay),
+            mp_obj_new_int(date.WeekDay - 1),
```

Comment:
This can't change, it's `pyb.RTC` which is in widespread use.

Instead we will need to split out `machine.RTC` as a different type to `pyb.RTC`, and make `machine.RTC` conform to this standard, but leave `pyb.RTC` alone.

Let's do that later, for now best to just leave stm32 as-is.

Reply:
Sure, I'll resubmit without the change to this file.

Reply:
looks good now, thanks!